go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go

```diff
@@ -34,6 +36,16 @@ type LogicalSwitchManager struct {
 	ipamFunc ipamFactoryFunc
 }
 
+// GetUUID returns the UUID for the given logical switch name if
+func (manager *LogicalSwitchManager) GetUUID(name string) (string, bool) {
+	manager.Lock()
```

Comment:
@JacobTanenbaum this could be an RLock()

Reply:
done